placeSpecificContent/hq/chatterboxes/chatterbox/buttons/privacyButton_server.js

```diff
@@ -0,0 +1,404 @@
+                    velocity: { x: 0, y: 0, z: 0 }
+                });
+                Entities.editEntity(button, {
+                    color: BUTTON_OPEN_COLOR
+                });
+                if (preloadDataRetrievalInterval) {
+                    Script.clearInterval(preloadDataRetrievalInterval);
+                    preloadDataRetrievalInterval = null;
+                }
+            } else if (preloadIntervalCount >= PRELOAD_DATA_RETRIEVAL_MAX_ATTEMPTS) {
+                if (preloadDataRetrievalInterval) {
+                    Script.clearInterval(preloadDataRetrievalInterval);
+                    preloadDataRetrievalInterval = null;
+                }
+            }
+        },
+
+        /* Use parenting relationships to get IDs of required entities */
+        getEntityData: function() {
+            if (!table) {
+                var table = Entities.getEntityProperties(_this.entityID, 'parentID').parentID;
+                Entities.getChildrenIDs(table).forEach(function(tableChildID) {
+                    var properties = Entities.getEntityProperties(tableChildID, ['position', 'name']);
+                    if (properties.name === "Chatterbox Privacy Button") {
+                        button = tableChildID;
+                        buttonPosition = properties.position;
+                        chatterbox = Entities.getEntityProperties(table, 'parentID').parentID;
+                    }
+                });
+            }
+            if (chatterbox && (!door || !shade)) {
+                Entities.getChildrenIDs(chatterbox).forEach(function(chatterboxChildID) {
+                    var properties = Entities.getEntityProperties(chatterboxChildID, ['position', 'name']);
+                    if (properties.name === "Chatterbox Door") {
+                        door = chatterboxChildID;
+                    } else if (properties.name === "Chatterbox Shade") {
+                        shade = chatterboxChildID;
+                    }
+                });
+            }
+        },
+
+        /* PLAY A SOUND: Plays a sound at the specified position, volume, local mode, and playback 
+        mode requested. */
+        playSound: function(sound, volume, position, localOnly, loop) {
+            if (sound.downloaded) {
+                if (injector) {
+                    injector.stop();
+                    injector = null;
+                }
+                injector = Audio.playSound(sound, {
+                    position: position,
+                    volume: volume,
+                    localOnly: localOnly,
+                    loop: loop
+                });
+            }
+        },
+
+        /* Toggle between between open and private chatterbox states.  */
+        transition: function() {
+            if (!ready || repeatTransitionCall) {
+                return;
+            }
+            /* When the client script calls ths function, it is being received multiple times despite only one event 
```

Comment:
this concerns me. i would prefer this code (and everything to do with `repeatTransitionCall`) to be removed from this code. can you please further investigate why this is necessary? i haven't seen this in previous pieces of content.

Reply:
I have seen this for a long time with mouse press calls. We can investigate but we need to let people know this project will not be finished soon then.